### DISCUSSION TITLE
Pi Pico W onboard LED is not connected to GPIO 25, connected to Wifi chip. . .

### DISCUSSION THREAD

**Discussion by marklacas** [tabemann/zeptoforth#23] [Q&A]
Anyone have any code to control the onboard LED on the pico W through the wifi chip?
MicroPython has solved this dilemma by using 'LED' for the pin.
Been trying to find the code, but failing.
Thanks,
-ml

**Comment by tabemann** [comment]
No, I do not have any code to solve this issue, as I have not started work on Pico W support (I own a Pico W, but it is still in its packaging). This will come along eventually, but currently I am working on interfacing with the ESP WiFi chip on the Wio RP2040.

**Reply by marklacas** [reply]
Turns out GPIO 0 on the Wifi chip is the on board LED.  So one only needs to get the SPI going to the wifi chip to gain access to it.

**Comment by marklacas** [comment]
Excellent! I just ordered one of those.

**Reply by tabemann** [reply]
Just one note - the Wio RP2040 I find is rather poorly documented, and I am still trying to figure out how the thing interfaces with its ESP. I found the documentation on the ESP-AT protocol used, but I still have not figured out everything because I have yet to get the thing to work. (You can currently load zeptoforth on it, but except for the onboard LED and the ESP WiFi chip it does work, though. Note, however, that compiling to flash is rather slow on it due to the particular external flash used. If you want to build zeptoforth from sources, I would highly suggest building on your Pico or Pico W, using `utils/download_uf2_image.sh` (note - not currently in a release tarball) to extract a UF2 image, and then flashing your Wio RP2040 with that.)

**Reply by znmeb** [reply]
I have two Arduino Nano RP2040 Connect boards. They have a WiFi module on them and the documentation is pretty good, at least for MicroPython and the Arduino IDE. They're a tad expensive though. I'm also probably going to get a [PicoADK ](https://hackaday.io/project/188665-picoadk-audio-development-kit )after I get the Pimoroni Audio module working. I'm waiting for Tindie to get the headers back in stock.

**Comment by tabemann** [comment]
I just found [source code](https://github.com/IsQianGe/rp2040-spi/blob/master/ports/rp2/wifi_spi.c) for the MicroPython driver support for the Wio RP2040's ESP radio. I git cloned it just in case it were to go away for any reason, as it is extremely helpful.
